### PR TITLE
 close issue #124 handle irregular case in Korean future tense

### PR DESCRIPTION
**Pull request by yjlim5**
This PR:
- adds test cases for irregular ㄷ consonant case in Future and Prepared tense
- fixes the logic to handle the case correctly in doFuture()
- plus, minor test code clean up

### CODE REVIEW

File: src/korean.js [hunk 1]
```diff
@@ -344,7 +344,10 @@ class Korean {
       case 8: // ㄹ
         return `${preStem}${combineSymbols(stem)} 거야`;
       case 7: // ㄷ
-        stem[stem.length - 1] = 8;
+        // This condition skips and thus handles the irregular case of ㅏㄷ like 닫다
+        if (stem[stem.length - 2] !== 0 || preStem) {
```
Comment (roycep) [inline src/korean.js:348]: If I am understanding correctly, this is saying, if the medial jamo is **not** ㅏ , then we replace the the final jamo with ㄹ. 

Shouldn't that replacement to ㄹ be for any medial jamo? Meaning we need to includeㅏwhich is 0 in the stem breakdown. This means the conditional `stem[stem.length - 2] !== 0` should not be used.

**Review by roycep — CHANGES_REQUESTED** [review]

File: src/korean.js (hunk 1 above)
Comment (songz) [reply src/korean.js:348]: You understood this code correctly. 

I think that the replacement to ㄹ should not be for any medial jamo when ending consonant is ㄷ. When medial jamo is ㅏ, We don't change the ending consonant.

Thanks for the observation, @yjlim5 can you clarify?

Comment (roycep) [reply src/korean.js:348]: Maybe I am reading in too much into an edge case maybe?

깨닫다 (to perceive, to realize) -> 깨달을 거예요

but then 

닫다(to close) -> 닫을 거예요

http://www.koreanwikiproject.com/wiki/%E3%84%B7_irregular_verbs

Comment (yjlim5) [reply src/korean.js:348]: @roycep and @songz, thanks for double-checking the logic!  
To clarify, this logic was implemented to handle the edge case correctly.  Here's how it works:

Replace the final jamo with ㄹ IF:
1. If The stem's medial jamo is not ㅏ
2. If the word has a prestem (then the medial jamo does not matter).

깨닫다 has a prestem = "깨" and the stem = "닫".  So, it gets the final jamo replaced to be ㄹ ==>  깨달을 거예요.

Let me know if any other question, and please review once more!

**Review by roycep — APPROVED** [review]
The behavior for the irregular ㄷ is correct with the prestem along with it!

**Comment by yjlim5** [comment]
Yay! Thanks again @roycep for your review, and @songz for catching this bug from Llip lessons.

**Pull request merged**